v2/pkg/release/graph.go

```diff
@@ -0,0 +1,83 @@
+package release
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+
+	"github.com/openshift/oc-mirror/v2/pkg/imagebuilder"
+)
+
+// createGraphImage creates a graph image from the graph data
+// and returns the image reference.
+// it follows https://docs.openshift.com/container-platform/4.13/updating/updating-restricted-network-cluster/restricted-network-update-osus.html#update-service-graph-data_updating-restricted-network-cluster-osus
+func (o *LocalStorageCollector) CreateGraphImage(ctx context.Context) (string, error) {
+	// HTTP Get the graph updates from api endpoint
+	resp, err := http.Get(graphURL)
+	if err != nil {
+		return "", err
+	}
+	defer resp.Body.Close()
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return "", err
+	}
+
+	// save graph data in a container layer modifying UID and GID to root.
+	archiveDestination := filepath.Join(o.Opts.Global.Dir, graphArchive)
+	graphLayer, err := imagebuilder.LayerFromGzipByteArray(body, archiveDestination, graphDataDir, 0644, 0, 0)
+	if err != nil {
+		return "", err
+	}
+	defer os.Remove(archiveDestination)
+
+	// Create a local directory for saving the OCI image layout of UBI9
+	layoutDir := filepath.Join(o.Opts.Global.Dir, graphPreparationDir)
+	if err := os.MkdirAll(layoutDir, os.ModePerm); err != nil {
+		return "", err
+	}
+
+	// Create an imageBuilder
+	nameOptions, remoteOptions := setImgBuilderDefaultOptions(ctx)
+	imgBuilder := imagebuilder.NewBuilder(nameOptions, remoteOptions, o.Log)
```

Comment:
Keeping the image builder in an image builder pkg is great, thanks! It will allow oc-mirror to reuse this package for other scenarios as well.